src/controllers/VueController.php

```diff
@@ -117,12 +77,16 @@ protected static function convertCurrency(float $amount, string $currency, bool
      */
     private function fetchOrders($id = null) : array
     {
-        $single = $_GET['purchasableId'] ?? $id;
-        $start  = $this->start_date ?: '1970-01-01 00:00:00';
-        $end    = $this->end_date ?: gmdate('Y-m-d 23:59:59');
-        $orders = Order::find()->dateOrdered(['and', ">= {$start}", "< {$end}"]);
-        $url    = $_SERVER['REQUEST_URI'];
-        $path   = basename(parse_url($url, PHP_URL_PATH));
+        $single   = $_GET['purchasableId'] ?? $id;
+        $start    = $this->start_date ? \DateTime::createFromFormat('Y-m-d H:i:s', $this->start_date) : '1970-01-01 00:00:00';
+        $end      = $this->end_date ? \DateTime::createFromFormat('Y-m-d H:i:s', $this->end_date) : gmdate('Y-m-d 23:59:59');
+        $numDays  = $end->diff($start)->format("%r%a");
+        $newStart = $start->modify($numDays . ' day')->format('Y-m-d 00:00:00');
```

Comment:
`modify`, `sub` and `add` mutate their DateTime instance and return that instance for chaining. The returned instance is not a new DateTime object. 

So in this case, `$newStart === $start`.

```suggestion
        $newStart = (clone $start)->modify($numDays . ' day')->format('Y-m-d 00:00:00');
```

or

```suggestion
        $newStart = (clone $start)->sub($start->diff($end))->format('Y-m-d 00:00:00');
```